tree/dataframe/src/RLoopManager.cxx

```diff
@@ -377,13 +376,46 @@ RLoopManager::RLoopManager(const ROOT::RDF::RDatasetSpec &spec)
       // Some other times, each different file has its own tree name, we need to
       // reconstruct the full path to the tree in each file and pass that to
       // TChain::Add
-      auto nfiles = spec.fFileNameGlobs.size();
-      for (decltype(nfiles) i = 0; i < nfiles; i++){
-         const auto fullpath = spec.fFileNameGlobs[i] + "?#" + spec.fSubTreeNames[i];
+      for (auto i = 0u; i < spec.fFileNameGlobs.size(); i++) {
+         const auto fullpath = spec.fFileNameGlobs[i] + "?#" + spec.fTreeNames[i];
          chain->Add(fullpath.c_str());
       }
    }
    SetTree(chain);
+
+   const auto &friendNames = fFriendInfo.fFriendNames;
```

Comment:
We need all the extra logic that attaches the friend info to the current chain that RDF is processing because previously we let the user first build the TTree+friends on their own and now they only specify it through the dataset spec. Right?

Reply:
this statement sounds right, yes. So it suffices to take care of the friend creation in the RLoopManager ctor.